### PR TITLE
Visitor can read a specific article 

### PR DESCRIPTION
**Pull request by 1CIM**
### User story

```
As a Visitor
In order to read an article
I would like to be able to click on an article and have its full content displayed
```

## Suggested Changes 

- Single page reading of selected article 
- Refactoring to use React Redux 
- Style changes and updates

## Dependencies 

- React Router Dom 
- React Redux 

## What we have learned 

- Usage of react Redux
- Usage of React Router Dom to move around the app 

## Screen Preview

https://recordit.co/qJ076leDBr

### CODE REVIEW

File: src/components/MainPage.jsx [hunk 1]
```diff
@@ -1,29 +1,30 @@
-import React, { useState, useEffect } from 'react';
+import React, { useEffect } from 'react';
 import { Header } from 'semantic-ui-react';
 import Articles from '../modules/Articles';
 import ArticleCard from './ArticleCard';
 import BreakingNews from './layout/BreakingNews';
+import { useSelector } from 'react-redux';
 
 const MainPage = () => {
```
Comment (DavDan1) [inline src/components/MainPage.jsx:8]: Looks cleaner with redux! nice!

File: src/components/MainPage.jsx [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
+    return <ArticleCard article={article} index={index} key={index} />;
   });
 
   return (
     <>
-      {errorMessage && 
+      {error && (
         <Header data-cy='error-message' color='red'>
-          {errorMessage}
+          {message}
```
Comment (DavDan1) [inline src/components/MainPage.jsx:23]: why is it called message not errorMessage?

File: src/modules/Articles.js [hunk 3]
```diff
@@ -1,20 +1,43 @@
...
-        );
-      } else {
-        setErrorMessage(error.message);
-      }
+      errorHandler(error);
+    }
+  },
+
+  async show(id) {
+    debugger;
```
Comment (DavDan1) [inline src/modules/Articles.js]: debugger :)

File: src/modules/Articles.js [hunk 4]
```diff
@@ -1,20 +1,43 @@
...
+    try {
+      const response = await axios.get(`/articles/${id}`);
+      debugger;
```
Comment (DavDan1) [inline src/modules/Articles.js]: please remove the debuggers

**Review by DavDan1 — APPROVED** [review]

File: src/modules/Articles.js (hunk 3 above)
Comment (1CIM) [reply src/modules/Articles.js]: was just to test you 😅 hehe 🙈

File: src/components/MainPage.jsx (hunk 2 above)
Comment (lukeperrin10) [reply src/components/MainPage.jsx:23]: With the Redux changes, the message appears when the error is true. I understand the confusion and we can look can probably change this going forward. With how the error messages are set up they will be used application wide.

File: src/modules/Articles.js (hunk 4 above)
Comment (1CIM) [reply src/modules/Articles.js]: both are removed now 👌

File: src/components/MainPage.jsx (hunk 2 above)
Comment (DavDan1) [reply src/components/MainPage.jsx:23]: Thanks for the explanation!

**Review by DavDan1 — APPROVED** [review]

**Review by DavDan1 — APPROVED** [review]

File: src/components/Article.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+          </h1>
+          <p data-cy='article-category' className='article-category'>
+            {article.category}
+          </p>
+          <p data-cy='article-body' className='article-body'>
+            {article.body}
+          </p>
+          <p data-cy='article-author' className='article-author'>
+            {article.author &&
+              article.author.first_name + ' ' + article.author.last_name}
```
Comment (offcast-dev) [inline src/components/Article.jsx]: string interpolation could be better choice

File: src/components/ArticleCard.jsx [hunk 6]
```diff
@@ -1,23 +1,29 @@
...
-          <Image src='https://images.unsplash.com/photo-1487758608033-7780b34680ac?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=3589&q=80' />
-          <div className='content'>
-            <h4 data-cy='title'>{article.title}</h4>
-            <Card.Description data-cy='teaser'>
+      <Link to={`/articles/${article.id}`}>
+        <div id={article.id}>
+          <img
+            className='card-image'
+            src='https://images.unsplash.com/photo-1487758608033-7780b34680ac?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=3589&q=80'
+            alt=''
```
Comment (offcast-dev) [inline src/components/ArticleCard.jsx]: why do you need  empty alt? Put something in there

File: src/components/layout/BreakingNews.jsx [hunk 7]
```diff
@@ -1,21 +1,30 @@
...
-            alt=''
-          />
-          <div className='breaking-content'>
-            <h2 data-cy='breaking-title' >{article.title}</h2>
-            <h4 data-cy='breaking-teaser' >{article.teaser}</h4>
-            <p data-cy='breaking-date' >{article.date}</p>
-          </div>
+          className='breaking-img'
+          src='https://images.unsplash.com/photo-1487758608033-7780b34680ac?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=3589&q=80'
+          alt=''
```
Comment (offcast-dev) [inline src/components/layout/BreakingNews.jsx]: same

File: src/state/store/configureStore.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { createStore } from 'redux';
+import rootReducer from '../reducers/rootReducers';
+import initialState from './initialState';
+
+const store = createStore(
+  rootReducer,
+  initialState,
+  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
```
Comment (offcast-dev) [inline src/state/store/configureStore.js:8]: why do you need this?

File: src/state/store/initialState.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+const initialState = {
+  error: false,
+  articles: [],
+  article: [],
```
Comment (offcast-dev) [inline src/state/store/initialState.js]: shound't it be a object?

**Review by offcast-dev — COMMENTED** [review]
Not critical but some stuff can be changed

File: src/state/store/configureStore.js (hunk 8 above)
Comment (1CIM) [reply src/state/store/configureStore.js:8]: This is for a chrome dev tool extension. Pretty nice if i say it myself.
Check it out here: https://github.com/zalmoxisus/redux-devtools-extension

File: src/components/Article.jsx (hunk 5 above)
Comment (lukeperrin10) [reply src/components/Article.jsx]: you are correct and change.

File: src/components/ArticleCard.jsx (hunk 6 above)
Comment (lukeperrin10) [reply src/components/ArticleCard.jsx]: fixed

File: src/components/layout/BreakingNews.jsx (hunk 7 above)
Comment (lukeperrin10) [reply src/components/layout/BreakingNews.jsx]: fixed

File: src/state/store/initialState.js (hunk 9 above)
Comment (lukeperrin10) [reply src/state/store/initialState.js]: fixed

File: src/state/store/configureStore.js (hunk 8 above)
Comment (offcast-dev) [reply src/state/store/configureStore.js:8]: Cool!

**Review by offcast-dev — APPROVED** [review]

**Pull request merged**